src/signature/commitment_signature.rs

```diff
@@ -0,0 +1,292 @@
+    #[inline]
+    pub fn get_public_commitment_nonce(&self) -> &HomomorphicCommitment<P> {
+        &self.public_commitment_nonce
+    }
+}
+
+impl<'a, 'b, P, K, C> Add<&'b CommitmentSignature<P, K, C>> for &'a CommitmentSignature<P, K, C>
+where
+    P: PublicKey<K = K>,
+    &'a HomomorphicCommitment<P>: Add<&'b HomomorphicCommitment<P>, Output = HomomorphicCommitment<P>>,
+    K: SecretKey,
+    &'a K: Add<&'b K, Output = K>,
+    C: HomomorphicCommitmentFactory<P = P> + Default,
+{
+    type Output = CommitmentSignature<P, K, C>;
+
+    fn add(self, rhs: &'b CommitmentSignature<P, K, C>) -> CommitmentSignature<P, K, C> {
+        let r_sum = self.get_public_commitment_nonce() + rhs.get_public_commitment_nonce();
+        let s_u_sum = self.get_signature_u() + rhs.get_signature_u();
+        let s_v_sum = self.get_signature_v() + rhs.get_signature_v();
+        CommitmentSignature::new(r_sum, s_u_sum, s_v_sum)
+    }
+}
+
+impl<'a, P, K, C> Add<CommitmentSignature<P, K, C>> for &'a CommitmentSignature<P, K, C>
+where
+    P: PublicKey<K = K>,
+    for<'b> &'a HomomorphicCommitment<P>: Add<&'b HomomorphicCommitment<P>, Output = HomomorphicCommitment<P>>,
+    K: SecretKey,
+    for<'b> &'a K: Add<&'b K, Output = K>,
+    C: HomomorphicCommitmentFactory<P = P> + Default,
+{
+    type Output = CommitmentSignature<P, K, C>;
+
+    fn add(self, rhs: CommitmentSignature<P, K, C>) -> CommitmentSignature<P, K, C> {
+        let r_sum = self.get_public_commitment_nonce() + rhs.get_public_commitment_nonce();
+        let s_u_sum = self.get_signature_u() + rhs.get_signature_u();
+        let s_v_sum = self.get_signature_v() + rhs.get_signature_v();
+        CommitmentSignature::new(r_sum, s_u_sum, s_v_sum)
+    }
+}
+
+impl<P, K, C> Default for CommitmentSignature<P, K, C>
+where
+    P: PublicKey<K = K>,
+    K: SecretKey,
+    C: HomomorphicCommitmentFactory<P = P> + Default,
+{
+    fn default() -> Self {
+        CommitmentSignature::new(HomomorphicCommitment::<P>::default(), K::default(), K::default())
+    }
+}
+
+/// Provide an efficient ordering algorithm for Commitment signatures. It's probably not a good idea to implement `Ord`
+/// for secret keys, but in this instance, the signature is publicly known and is simply a scalar, so we use the hex
+/// representation of the scalar as the canonical ordering metric. This conversion is done if and only if the public
+/// nonces are already equal, otherwise the public nonce ordering determines the CommitmentSignature order.
+impl<P, K, C> Ord for CommitmentSignature<P, K, C>
+where
+    P: PublicKey<K = K>,
+    K: SecretKey,
+    C: HomomorphicCommitmentFactory<P = P> + Default,
+{
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self
+            .get_public_commitment_nonce()
+            .cmp(&other.get_public_commitment_nonce())
+        {
+            Ordering::Equal => {
+                let this_u = self.get_signature_u().to_hex();
```

Comment:
Is the hex conversion necessary here?

Reply:
its either a hex conversion or we need to impl ord with secret key

Reply:
surely the hex has the same order as sorting the bytes?